domain/model/types.go

```diff
@@ -177,3 +179,35 @@ func WithDeleteDB() DeleteModelOption {
 		o.deleteDB = true
 	}
 }
+
+// StatusInfo holds a Status and associated information of a model.
+type StatusInfo struct {
+	// Status is the current status of the model.
+	Status corestatus.Status
+	// Message is a human-readable message that describes the current status of the model.
+	Message string
+	// Reason is a human-readable message that describes the reason for the current status of the model.
+	Reason string
+	// Since is the time when the model entered the current status.
+	Since *time.Time
```

Comment:
In an ideal world, if we could go back in time, we'd not allow a nil value since surely it's an expectation that the time a status value is set is known and should be modelled. The reasons for allowing nil might be lost in time; it could be test leakage since I know we have tests that do not assert on the timestamp if it is nil etc. Although it would be nice to fix past mistakes, we have to be very careful about subtle api breakage. In params structs like StatusDetails we set the status timestamp as a pointer directly from the underlying (nilable) status value. So the caller might be placing behaviour on receiving such a nil value. I'm not aware of any documentation around this. It might be a "safe" change but my current thoughts are the safest course of action would be to retain current behaviour. There's so many potential client breakages - libjuju, dashboard etc

Reply:
The agreed direction is that we we transition to a status timestamp that is not a pointer. We will deal with any client breakages that result. Please change this to a `time.Time`. We can punt on `time.Now()` for those places where timestamp is not known but will need a plan to fix.